### ISSUE TITLE
How to use mmslfesup to train resnet50 caffe pre-training weights in mmdetection

### ISSUE THREAD

**Issue by weikai520** [open-mmlab/mmselfsup#546]
### What is the problem this feature will solve?

I saw the weight of resnet50 caffe.pth in mmdetection, can we use self-supervised model to train him
![image](https://user-images.githubusercontent.com/86449151/197507913-1326f435-cdde-4c79-a2fd-d59cab493c24.png)
Do I only need to modify the style in resnet.py in mmselfsup？
![image](https://user-images.githubusercontent.com/86449151/197508329-8c463739-a8ad-4088-8768-a7cf692149f2.png)

### What is the feature?

I saw the weight of resnet50 caffe.pth in mmdetection, can we use self-supervised model to train him

### What alternatives have you considered?

Do I only need to modify the style in resnet.py in mmselfsup？

**Comment by fangyixiao18** [comment]
Hello, the style need to be changed. Besides, you could try to load the checkpoint to see whether it would report the 'unexpected_keys' or 'missing_keys' in the log.

**Comment by weikai520** [comment]
> 您好，样式需要更改。此外，您可以尝试加载检查点以查看它是否会在日志中报告“unexpected_keys”或“missing_keys”。

Yes, when I download resnet50_caffe.pth and use 'load_from' he will report the 'unexpected_keys' or 'missing_keys' in the log

**Comment by weikai520** [comment]
Do you have any good solution or can you solve it?

**Comment by fangyixiao18** [comment]
You can check the part of 'unexpected_keys' or 'missing_keys', the backbone, the neck, or the head?

And generally, we first train self-supervised learning and then transfer the backbone weights to some downstream task such as detection

**Comment by weikai520** [comment]
> 您可以检查“unexpected_keys”或“missing_keys”的部分，骨干，颈部或头部？
> 
> 一般来说，我们首先训练自我监督学习，然后将骨干权重转移到一些下游任务，如检测。

I understand what you mean, but resnet50_caffe is also a pre-trained weight, can't it load the training? The weight I downloaded is only the backbone network resnet50, no head and neck.

**Comment by weikai520** [comment]
> 您可以检查“unexpected_keys”或“missing_keys”的部分，骨干，颈部或头部？
> 
> 一般来说，我们首先训练自我监督学习，然后将骨干权重转移到一些下游任务，如检测。

![image](https://user-images.githubusercontent.com/86449151/197516647-d2f7c8f6-14d8-4bf0-a9b3-e729da0868a0.png)

**Comment by fangyixiao18** [comment]
> I understand what you mean, but resnet50_caffe is also a pre-trained weight, can't it load the training? The weight I downloaded is only the backbone network resnet50, no head and neck.

It can be loaded, you could copy-paste the 'unexpected_keys' and 'missing_keys' here.
And if there are some different keys, you could modify the keys by yourself to ensure that the weights are loaded correctly

**Comment by weikai520** [comment]
> > I understand what you mean, but resnet50_caffe is also a pre-trained weight, can't it load the training? The weight I downloaded is only the backbone network resnet50, no head and neck.
> 
> It can be loaded, you could copy-paste the 'unexpected_keys' and 'missing_keys' here. And if there are some different keys, you could modify the keys by yourself to ensure that the weights are loaded correctly

There are many different 'keys', and almost all of them cannot be matched. I don't know what to do. How can I view the 'key' of the 'pth' file

**Comment by fangyixiao18** [comment]
Here is a simple example:
```python
import torch

path = 'xxxxxxx.pth'
ckpt = torch.load(path)

print(ckpt['state_dict'].keys())
```
